api/src/main/java/marquez/service/LineageService.java

```diff
@@ -48,14 +49,27 @@ public LineageService(LineageDao delegate, JobDao jobDao) {
 
   // TODO make input parameters easily extendable if adding more options like 'withJobFacets'
   public Lineage lineage(NodeId nodeId, int depth, boolean withRunFacets) {
+    log.debug("Attempting to get lineage for node '{}' with depth '{}'", nodeId.getValue(), depth);
     Optional<UUID> optionalUUID = getJobUuid(nodeId);
     if (optionalUUID.isEmpty()) {
-      throw new NodeIdNotFoundException("Could not find node");
+      throw new NodeIdNotFoundException(String.format("Node '%s' not found!", nodeId.getValue()));
     }
     UUID job = optionalUUID.get();
-
+    log.debug("Attempting to get lineage for job '{}'", job);
     Set<JobData> jobData = getLineage(Collections.singleton(job), depth);
 
+    // Ensure job data is not empty, an empty set cannot be passed to LineageDao.getCurrentRuns() or
+    // LineageDao.getCurrentRunsWithFacets().
+    if (jobData.isEmpty()) {
+      // Log warning, then return an empty lineage graph; a graph must contain at most one
+      // job->dataset relationship.
+      log.warn(
+          "Failed to get lineage for job '{}' associated with node '{}', returning empty graph...",
+          job,
+          nodeId.getValue());
+      return toLineage(Sets.newHashSet(), Sets.newHashSet());
```

Comment:
Isn't the graph in the UI re-rendered based on the contents of this response? If a dataset is selected, rather than a graph with a single node, we'll show an error, right? We should make sure that the node is a valid node (e.g., a real dataset), then respond with at least that node in the graph.